framework/libs/id_allocator/src/lib.rs

```diff
@@ -145,6 +148,26 @@ impl IdAlloc {
         }
     }
 
+    /// Reserve a specific `id` from the available `id`s.
+    ///
+    /// If the id is already allocated, it returns false, otherwise true.
+    ///
+    /// # Panic
+    ///
+    /// If the `id` is out of bounds, this method will panic.
+    pub fn reserve(&mut self, id: usize) -> bool {
```

Comment:
I think this method should be named `alloc_target` or `alloc_specific` in order to be aligned with `alloc` and `alloc_consecutive`.

In addition, I often find the semantics of a method returning boolean confusing, unless the method is named `is_xxx` or `can_xxx`. Without reading the document, people may be confused as to whether the meaning of a `true` returned from the `reserve` method: the id is _already allocated_ or _reserved successfully_? This ambiguity can be avoid by naming it `alloc_specific` and make it return `Option<usize>` or `Option<()>`. As the method means to allocate, it is obvious that `Some(_)` means a successful allocation, whereas a `None` means a failure.